presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcBatchPageSourceFactory.java

```diff
@@ -268,7 +269,9 @@ public static ConnectorPageSource createOrcPageSource(
             }
 
             if (!physicalColumns.isEmpty() && physicalColumns.stream().allMatch(hiveColumnHandle -> hiveColumnHandle.getColumnType() == AGGREGATED)) {
-                return new AggregatedOrcPageSource(physicalColumns, reader.getFooter(), typeManager, functionResolution);
+                Footer footer = reader.getFooter();
+                orcDataSource.close();
```

Comment:
OrcDataSouce implements Closeable so try-with-resources would give even stronger guarantees that the object is closed along all paths. This would require a little more refactoring of the flow control to use larger scoped try blocks.

Reply:
try-with-resource doesn't really help in this case as the resource lifecycle does not align with the try-with-resource idiom.

A slightly better approach would be to pass the reader to the aggregating page source, but it won't work without additional changes because:
- 1. OrcReader is not Closeable, and generally the AbstractOrcRecordReader controls the data source.
- 2. the aggregating page source only needs the file footer, so closing the OrcDataSouce early makes sense to release memory as early as possible given that Presto is usually constrained on memory.

To be honest, I don't see much value in doing a refactoring for a single line change.